test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTemplateEngineFactoryServiceTest.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Host;
+using Mvc1_X = Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X;
+using MvcLatest = Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class DefaultTemplateEngineFactoryServiceTest
+    {
+        [Fact]
+        public void Create_CreatesDesignTimeTemplateEngine_ForLatest()
+        {
+            // Arrange
+            var mvcReference = GetAssemblyMetadataReference("Microsoft.AspNetCore.Mvc.Razor", "2.0.0");
+            var services = GetServices(mvcReference);
+            var factoryService = new DefaultTemplateEngineFactoryService(services);
+
+            // Act
+            var engine = factoryService.Create("/TestPath/SomePath/", b =>
+            {
+                b.Features.Add(new MyCoolNewFeature());
+                Assert.True(b.DesignTime);
+            });
+
+            // Assert
+            Assert.Single(engine.Engine.Features.OfType<MyCoolNewFeature>());
+            Assert.Single(engine.Engine.Features.OfType<MvcLatest.MvcViewDocumentClassifierPass>());
```

Comment:
I decided not to extensively test everything in the Engine and instead assert a few features that are relevant.